client/landing/stepper/hooks/use-save-hosting-flow-path-step.ts

```diff
@@ -0,0 +1,19 @@
+import { IMPORT_HOSTED_SITE_FLOW, isAnyHostingFlow } from '@automattic/onboarding';
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'calypso/state';
+import { getCurrentUserId, isCurrentUserEmailVerified } from 'calypso/state/current-user/selectors';
+import { savePreference } from 'calypso/state/preferences/actions';
+
+export function useSaveHostingFlowPathStep( flow: string | null, currentPath: string ) {
+	const dispatch = useDispatch();
+	const userId = useSelector( getCurrentUserId );
+	const isEmailVerified = useSelector( isCurrentUserEmailVerified );
+	const pathStep = isEmailVerified ? null : currentPath;
+
+	useEffect( () => {
+		if ( ! isEmailVerified && ( isAnyHostingFlow( flow ) || flow === IMPORT_HOSTED_SITE_FLOW ) ) {
+			const prefKey = `hosting-flow-path-step-${ userId }`;
```

Comment:
Would this be useful for any onboarding flow that might be segmented with an email verification step? If so we could update the name of the hook and prefKey to be more general to onboarding (as well as on the diff).

Reply:
It might be, I'm not familiar with the other flows to be honest. If this comes up again (on other flows) we can always refactor this hook to be more generic.